File: app/src/main/java/com/dumuzeyn/mp3player/NowPlayingIndicator.java
package com.dumuzeyn.mp3player;

import android.content.Context;
import android.graphics.drawable.GradientDrawable;
import android.view.Gravity;
import android.view.View;
import android.widget.FrameLayout;

/** Creates the shared current-track marker used by every library surface. */
final class NowPlayingIndicator {
    private NowPlayingIndicator() {
    }

    static View create(MainActivityCore host) {
        View indicator = new View(host);
        style(indicator, host.yellow);
        return indicator;
    }

    static void style(View indicator, int color) {
        GradientDrawable background = new GradientDrawable();
        background.setColor(color);
        background.setCornerRadius(indicator.getResources().getDimension(
                R.dimen.now_playing_indicator_width));
        indicator.setBackground(background);
    }

    static FrameLayout.LayoutParams layoutParams(Context context) {
        int width = size(context, R.dimen.now_playing_indicator_width);
        int height = size(context, R.dimen.now_playing_indicator_height);
        int inset = size(context, R.dimen.now_playing_indicator_inset);
        FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(width, height);
        params.gravity = Gravity.START | Gravity.CENTER_VERTICAL;
        params.setMargins(inset, 0, 0, 0);
        return params;
    }

    private static int size(Context context, int resource) {
        return context.getResources().getDimensionPixelSize(resource);
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/NowPlayingIndicator.kt
package com.dumuzeyn.mp3player

import android.content.Context
import android.graphics.drawable.GradientDrawable
import android.view.Gravity
import android.view.View
import android.widget.FrameLayout

/** Creates the shared current-track marker used by every library surface. */
object NowPlayingIndicator {
    @JvmStatic
    fun create(host: MainActivityCore): View = View(host).also { style(it, host.yellow) }

    @JvmStatic
    fun style(indicator: View, color: Int) {
        indicator.background = GradientDrawable().apply {
            setColor(color)
            cornerRadius = indicator.resources.getDimension(R.dimen.now_playing_indicator_width)
        }
    }

    @JvmStatic
    fun layoutParams(context: Context): FrameLayout.LayoutParams {
        val width = size(context, R.dimen.now_playing_indicator_width)
        val height = size(context, R.dimen.now_playing_indicator_height)
        val inset = size(context, R.dimen.now_playing_indicator_inset)
        return FrameLayout.LayoutParams(width, height).apply {
            gravity = Gravity.START or Gravity.CENTER_VERTICAL
            setMargins(inset, 0, 0, 0)
        }
    }

    private fun size(context: Context, resource: Int): Int =
        context.resources.getDimensionPixelSize(resource)
}
